distage/distage-testkit-scalatest/src/main/scala-2/izumi/distage/testkit/scalatest/AssertCIO.scala

```diff
@@ -22,6 +38,42 @@ object AssertCIO extends AssertCIO {
       import c.universe._
       c.Expr[IO[Assertion]](q"_root_.cats.effect.IO.delay(${DistageAssertionsMacro.assert(c)(arg)(prettifier, pos)})")
     }
+
+    def shortImpl1[T: c.WeakTypeTag](
+      c: blackbox.Context
+    )(effect: c.Expr[IO[T]]
+    )(predicate: c.Expr[T => Boolean]
+    )(prettifier: c.Expr[Prettifier],
+      pos: c.Expr[Position],
+    ): c.Expr[IO[Assertion]] = {
+      import c.universe._
+
+      val resultName = TermName(c.freshName("result"))
+      val predicateBody = ReflectionUtil.betaReduceLambda(c)(predicate, List(resultName.toString))
+
+      c.Expr[IO[Assertion]](q"$effect.flatMap { ($resultName: ${weakTypeOf[T]}) => assertIO($predicateBody)($prettifier, $pos) }")
```

Comment:
@YaroslavSagach 
This reference to `assertIO` is ambiguous, it can resolve to `this.assertIO`, but also to any `assertIO` in scope of the macro expansion - it's not hygienic. Also, since this adds a nested macro expansion to the tree, which is not performant. (After this tree goes back to the typer, the typer will see an unexpanded macro assertIO and will rerun the macro expansion + typing cycle again, nested macros can be very deadly to compilation performance.)

Instead, run the regular assertIO macro (`impl` method) directly from inside this macro:

```suggestion
      c.Expr[IO[Assertion]](q"$effect.flatMap { ($resultName: ${weakTypeOf[T]}) => ${ impl(c)(predicateBody)(prettifier, pos) } }")
```

```suggestion
      c.Expr[IO[Assertion]](q"$effect.flatMap { ($resultName: ${weakTypeOf[T]}) => assertIO($predicateBody)($prettifier, $pos) }")
```